langchain/callbacks/manager.py

```diff
@@ -259,12 +262,17 @@ async def _ahandle_event(
     **kwargs: Any,
 ) -> None:
     """Generic event handler for AsyncCallbackManager."""
+    for handler in [h for h in handlers if h.run_inline]:
```

Comment:
Is that the same as 

```
_handle_event(sync_handlers)
```

And then we don't have to handle that case in _ahandle_event_for_handler

Reply:
Isn't the logic exactly the same in these two functions?

Reply:
but then we'd be repeating here the check for whether an event handler is sync or async?

Reply:
If we put the check here and call _handle_event on sync handlers, we don't need to change the _ahandle... function at all.

Sync handlers will never get there.

Reply:
But someone can set this run_inline property on an async handler? There is no easy way to prevent that.

Reply:
We cannot send an async handler into _handle_event

Reply:
it won't have any effect, as the concept doesn't make sense. But not sure its worth throwing an exception for it?

Reply:
I'm looking at the current implementation and see that we're not awaiting if `run_inline` is true. What will happen if we run async function with `run_inline` and don't await on it?

Reply:
<img width="644" alt="Screenshot 2023-06-19 at 15 39 19" src="https://github.com/hwchase17/langchain/assets/56902/72384909-17a7-4f25-bc9b-ba27ce154996">

In that branch we already know it's not async, look at the if statement above

Reply:
Ah yes, makes total sense!